server/src/services/diaries_service.ts

```diff
@@ -140,8 +141,19 @@ export const create = async (body: Diary) => {
     const createdDiaryTransaction = await model.sequelize.transaction(async (transaction) => {
       body.tempSentenceId = body.sentenceId;
       const diaryInfo = await model.Diary.create(body, { transaction });
-      // 글 작성후 새로운 문장 추천을 받기위함
-      await model.UsersRecommendedSentences.destroy({ where: { userId: body.userId }, transaction });
+
+      // 기존 추천문장 조회
+      const recommendedSentencesInfo: UsersRecommendedSentences[] = await model.UsersRecommendedSentences.findAll({
+        where: { userId: body.userId },
+      });
+
+      // 글 작성후 새로운 문장 추천을 받기위해 기존 추천문장 삭제
+      recommendedSentencesInfo.forEach(
+        async (element) => {
+          await element.destroy();
+        },
+        { transaction }
+      );
```

Comment:
이부분 내가 알기로는 await 이 비동기로 돌아서 제대로 동기적으로 수행 안되는걸로 알고있는데
map이나 foreach의 비동기처리 한번 검색해보면 좋겠네요!

Reply:
그리고 쿼리도 조금 확인 해봐야할듯! 조금 이상항!
```suggestion
      recommendedSentencesInfo.forEach(
        async (element) => {
          await element.destroy({ transaction });
        },
      );
```
이렇게 transaction옵션이 안에 들어가 있어야 하지 않을까

Reply:
그렇게 해보니까 에러가뜨더라고 그래서 저렇게 바꿨는데 되가지고 우선 저렇게 했는데..!

Reply:
아마아 트랜잭션이 안먹을텐데.......